### PR TITLE
Rework

### PR DESCRIPTION
**Pull request by DotWith**
# Major Changes
Added hxformat.
getFont returns name and size. Removed getFontSize.

# TODO
- [x] [Rework init](https://github.com/larsiusprime/firetongue/issues/43)
- [x] Formatted and added hxformat

### CODE REVIEW

**Comment by larsiusprime** [comment]
I can merge this when you're done, but since it looks like it has breaking changes we'll have to bump the version number a full point when it's done.

What's the main thing you're trying to achieve here?

**Comment by DotWith** [comment]
> I can merge this when you're done, but since it looks like it has breaking changes we'll have to bump the version number a full point when it's done.
> 
> What's the main thing you're trying to achieve here?

I'm attempting to make the library more easier to use.

**Comment by larsiusprime** [comment]
Cool! What are the chief improvements in this PR?

**Comment by DotWith** [comment]
> Cool! What are the chief improvements in this PR?

I need to add backwards compatibly with the old init function and getFontSize. I mark them as deprecated.

Init now uses params
```hx
tongue.init({
        locale: "en-US",
        finishedCallback: onLoaded
    });
```

getFontSize has been removed and instead replaced with this.
```hx
var oldfont = "verdana";
var oldsize = 12;
var newfont = tongue.getFont(oldfont, oldsize);
trace(newfont.name); // returns "comicsans"
trace(newfont.size); // returns 14
```

**Comment by larsiusprime** [comment]
Ah, that's much more convenient. Looks good, thanks for doing this!

**Comment by DotWith** [comment]
> Ah, that's much more convenient. Looks good, thanks for doing this!

I was hoping to add other property's like position, etc. To the FontData, but I haven't found a reason to add it so far.
```xml
<?xml version="1.0" encoding="utf-8" ?>
<data>
	<font value="verdana" replace="arial">
            <x value="13" replace="199"/>
            <y value="18" replace="456"/>
	</font>
</data>
```

**Comment by DotWith** [comment]
I'll add xml support into another pull request.

**Comment by DotWith** [comment]
@larsiusprime You shouldn't have to bump up the version to a full point anymore. Now that **backwards compatibly** has been added. Once you do merge, please update the `CHANGELOG.md`.

**Comment by larsiusprime** [comment]
Cool, let me know when you're done!

**Comment by DotWith** [comment]
@larsiusprime I'm done. I've tested the example project. This pr will close #40 and #43.

File: firetongue/FireTongue.hx
Comment (EliteMasterEric) [inline firetongue/FireTongue.hx:139]: It would be nice if this function could be reworked to use another parameter object, including a `customFramework` class instead of individual functions, but that should probably be postponed to the next point where a major version is deemed necessary.

**Review by EliteMasterEric — APPROVED** [review]
@larsiusprime For reference, I collaborated with DotWith on this, and made sure that the improvements they made were backwards compatible, so they should work without interfering with existing projects.

**Comment by larsiusprime** [comment]
> @larsiusprime For reference, I collaborated with DotWith on this, and made sure that the improvements they made were backwards compatible, so they should work without interfering with existing projects.

Happy to make you a collaborator on this repo too, but I'll try to rule on this PR by end of day nonetheless.

**Review by larsiusprime — APPROVED** [review]
This looks good!

**Pull request merged**
